unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BuilderInWorldEntityHandler.cs

```diff
@@ -709,6 +710,7 @@ public void DeleteEntity(DCLBuilderInWorldEntity entityToDelete, bool checkSelec
         hudController?.RefreshCatalogAssetPack();
         EntityListChanged();
         builderInWorldBridge?.RemoveEntityOnKernel(idToRemove, sceneToEdit);
+        OnEntityDeleted?.Invoke(entityToDelete.rootEntity.entityId);
```

Comment:
I think this event should be before the line 787, since we destroy the entity gameObject it can be null, also, if someone want to do changes on the entity before it gets finally delete it should be able to do it!

Reply:
Ok, done in https://github.com/decentraland/unity-renderer/pull/358/commits/ea850879045eb077e80380241672a91a48801b4b